### PR TITLE
Columbus-5 update

### PR DESCRIPTION
**Pull request by MSNTCS**

### CODE REVIEW

**Review by MSNTCS — COMMENTED** [review]
@ezaanm Please resolve conflicts.

**Review by YunSuk-Yeo — CHANGES_REQUESTED** [review]
Need some changes in raw query

File: contracts/market/src/contract.rs [hunk 1]
```diff
@@ -517,24 +567,24 @@ pub fn query_epoch_state<S: Storage, A: Api, Q: Querier>(
     })
 }
 
-pub fn migrate<S: Storage, A: Api, Q: Querier>(
-    deps: &mut Extern<S, A, Q>,
-    env: Env,
-    msg: MigrateMsg,
-) -> MigrateResult {
+#[cfg_attr(not(feature = "library"), entry_point)]
```
[the remark this answers is not in the document]
Comment (ezaanm) [reply contracts/market/src/contract.rs:570]: https://github.com/Anchor-Protocol/money-market-contracts/pull/39

File: contracts/market/src/migration.rs [hunk 2]
```diff
@@ -33,11 +33,11 @@ struct LegacyState {
     pub anc_emission_rate: Decimal256,
 }
 
-fn read_legacy_config<S: Storage>(storage: &S) -> StdResult<LegacyConfig> {
+fn read_legacy_config(storage: &dyn Storage) -> StdResult<LegacyConfig> {
```
[the remark this answers is not in the document]
Comment (ezaanm) [reply contracts/market/src/migration.rs:36]: https://github.com/Anchor-Protocol/money-market-contracts/pull/39

File: contracts/overseer/src/contract.rs [hunk 3]
```diff
@@ -518,58 +543,56 @@ pub fn query_config<S: Storage, A: Api, Q: Querier>(
...
     }
 }
 
-pub fn migrate<S: Storage, A: Api, Q: Querier>(
-    deps: &mut Extern<S, A, Q>,
-    _env: Env,
-    msg: MigrateMsg,
-) -> MigrateResult {
-    let config: Config = read_config(&deps.storage)?;
+#[cfg_attr(not(feature = "library"), entry_point)]
```
[the remark this answers is not in the document]
Comment (ezaanm) [reply contracts/overseer/src/contract.rs:585]: https://github.com/Anchor-Protocol/money-market-contracts/pull/39

File: packages/moneymarket/src/common.rs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+use cosmwasm_std::{Addr, Api, StdResult};
+
+pub fn optional_addr_validate(api: &dyn Api, addr: Option<String>) -> StdResult<Option<Addr>> {
```
[the remark this answers is not in the document]
Comment (ezaanm) [reply packages/moneymarket/src/common.rs:3]: nowhere else in this repo

**Comment by MSNTCS** [comment]
Here is alpha version of `moneymarket` https://crates.io/crates/moneymarket/0.3.0-alpha.1

**Comment by MSNTCS** [comment]
@ezaanm @YunSuk-Yeo I think we can merge this PR.

**Comment by ezaanm** [comment]
@MSNTCS @YunSuk-Yeo 
giphy

**Review by YunSuk-Yeo — APPROVED** [review]
lego

**Pull request merged**